components/sup/src/main.rs

```diff
@@ -181,7 +181,7 @@ fn sub_run(m: &ArgMatches, launcher: LauncherCli) -> Result<()> {
                     &msg.bldr_channel
                         .clone()
                         .map(ChannelIdent::from)
-                        .unwrap_or_default(),
+                        .unwrap_or_default(), // or configured_value?
```

Comment:
Looking at the information flow in this function, I don't think we can ever really end up with a `None` here; see `update_svc_load_from_input`.

Reply:
Based on [`update_svc_load_from_input`](https://github.com/habitat-sh/habitat/blob/master/components/sup/src/main.rs#L517) and its call to [`channel`](https://github.com/habitat-sh/habitat/blob/master/components/sup/src/main.rs#L361), I think you're right. The rework to get the complier to prove that is a bit much right now, so I'll change this to:
```suggestion
                        .expect("update_svc_load_from_input to always set to Some"),
```